packages/client/src/runtime/utils/validatePrismaClientOptions.ts

```diff
@@ -100,7 +106,7 @@ It should have this form: { url: "CONNECTION_STRING" }`,
       )
     }
 
-    if (getClientEngineType() === ClientEngineType.Binary) {
+    if (getClientEngineType(config.generator) === ClientEngineType.Binary) {
```

Comment:
ℹ️ Actually fixing a bug here I think: It always fell through to the env vars or default value and fully ignored the values in the generator block. Likely not a big issue though as `library` was the default anyways.